.travis.yml

```diff
@@ -1,12 +1,62 @@
+sudo: required
+dist: trusty
 language: python
+matrix:
+    include:
+        - python: 2.7
+          env: KERAS_BACKEND=theano TEST_MODE=PEP8
+        - python: 2.7
+          env: KERAS_BACKEND=theano
+        - python: 3.5
+          env: KERAS_BACKEND=theano
 
-python:
-        - 3.6
-        - 2.7
-
-script:
+install:
         - set -e
-        - pip install numpy theano keras
+        - pip install numpy theano
+        - pip install git+git://github.com/fchollet/keras.git
+        - pip install pep8 pytest-pep8
         - python setup.py install
         - KERAS_BACKEND=theano pytest
+install:
+  # code below is taken from http://conda.pydata.org/docs/travis.html
+  # We do this conditionally because it saves us some downloading if the
+  # version is the same.
+  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
+      wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh;
+    else
+      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+    fi
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - export PATH="$HOME/miniconda/bin:$PATH"
+  - hash -r
+  - conda config --set always_yes yes --set changeps1 no
+  - conda update -q conda
+  # Useful for debugging any issues with conda
+  - conda info -a
+
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy scipy matplotlib pandas pytest h5py
+  - source activate test-environment
+  #- pip install pytest-cov python-coveralls pytest-xdist coverage==3.7.1 #we need this version of coverage for coveralls.io to work
+  - pip install pep8 pytest-pep8
+  - pip install theano
+  - pip install git+git://github.com/fchollet/keras.git
 
+  - python setup.py install
+
+  # install TensorFlow (CPU)
+  # - pip install tensorflow
+
+# command to run tests
+script:
+  # run keras backend init to initialize backend config
+  - python -c "import keras.backend"
+  # create dataset directory to avoid concurrent directory creation at runtime
+  - mkdir ~/.keras/datasets
+  # set up keras backend
+  - sed -i -e 's/"backend":[[:space:]]*"[^"]*/"backend":\ "'$KERAS_BACKEND'/g' ~/.keras/keras.json;
+  - echo -e "Running tests with the following config:\n$(cat ~/.keras/keras.json)"
+  - if [[ "$TEST_MODE" == "PEP8" ]]; then
+       PYTHONPATH=$PWD:$PYTHONPATH py.test --pep8 -m pep8 -n0;
+    else
+       PYTHONPATH=$PWD:$PYTHONPATH py.test tests/;
```

Comment:
I think you need to call `pytest` (or maybe `py.test` works too) in the root directory of the repo. The tests aren't in a dedicated global `tests/` directory, but rather local to where the functionality is actually coded, as e.g. sklearn does it